### ISSUE TITLE
Crash in Open/Save dialog with dark themes

### ISSUE THREAD

**Issue by astromind** [gitextensions/gitextensions#8668]
## Current behaviour
System.NullReferenceException
Object reference not set to an instance of an object.

## Expected behaviour
No crash

## Steps to reproduce
Set theme to "dark" or "darksliver". (Restart to apply).

Open repository (Ctrl+O) - Browse - in the Windows "Select Folder" dialog click on the search box in the top-right corner -> crash.
Happens with other Open/Save dialogs as well.

## Error Details
```
System.NullReferenceException: Object reference not set to an instance of an object.
```

## Additional information
Works fine with "default" (white) theme.

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.13.2.windows.1 (recommended: 2.25.1 or later)
- Microsoft Windows NT 10.0.14393.0
- .NET Framework 4.6.1648.0
- DPI 120dpi (125% scaling)

**Comment by RussKie** [comment]
@NikolayXHD would you be able to have a look at it?

**Comment by gerhardol** [comment]
So this?
![image](https://user-images.githubusercontent.com/6248932/101953391-8ee27280-3bfa-11eb-88f3-455e0f726a2d.png)

Complete exception?

Works for me
Crash is in WinForms so back to @RussKie ?
Quite old Win and .NET version

**Comment by NikolayXHD** [comment]
Sorry for late reply, will look at the issue the next weekend.

**Comment by astromind** [comment]
@gerhardol I updated the first post with other environment and screenshots.

**Comment by gerhardol** [comment]
Can you copy the exception in the NBug window?
(Left bottom button.) That has the stack trace.

**Comment by astromind** [comment]
"Copy" button in NBug doesn't seem to have a stacktrace for me.

<details>
  <summary>Copy in NBug</summary>

```
<!--
    :warning: Review existing issues to see whether someone else has already reported your issue.
-->

:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Current behaviour

<!-- Be as specific and detailed as possible to help us identify your issue. -->


## Expected behaviour


## Steps to reproduce

<!-- Take some time to try and reproduce the issue, then explain how to do so here. -->


## Error Details
```
System.NullReferenceException: Object reference not set to an instance of an object.
```


## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.29.2.windows.3
- Microsoft Windows NT 10.0.17763.0
- .NET Framework 4.7.3690.0
- DPI 96dpi (no scaling)

```
</details>

<details>
  <summary>Stack trace from attached VS</summary>

```
 	[Managed to Native Transition]	
>	mscorlib.dll!System.Environment.Exit(int exitCode)	Unknown
 	GitExtensions.exe!GitExtensions.Program.ReportBug(System.Exception ex) Line 349	C#
 	GitExtensions.exe!GitExtensions.Program.Main.AnonymousMethod__1_1(object s, System.UnhandledExceptionEventArgs e) Line 50	C#
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	Microsoft.WindowsAPICodePack.Shell.dll!Microsoft.WindowsAPICodePack.Dialogs.CommonFileDialog.ShowDialog()	Unknown
 	Microsoft.WindowsAPICodePack.Shell.dll!Microsoft.WindowsAPICodePack.Dialogs.CommonFileDialog.ShowDialog(System.IntPtr ownerWindowHandle)	Unknown
 	GitUI.dll!GitUI.OsShellUtil.PickFolder(System.Windows.Forms.IWin32Window ownerWindow, string selectedPath) Line 57	C#
 	GitUI.dll!GitUI.CommandsDialogs.BrowseDialog.FormOpenDirectory.folderBrowserButton_Click(object sender, System.EventArgs e) Line 118	C#
 	System.Windows.Forms.dll!System.Windows.Forms.Control.OnClick(System.EventArgs e)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Button.OnClick(System.EventArgs e)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Button.OnMouseUp(System.Windows.Forms.MouseEventArgs mevent)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.WmMouseUp(ref System.Windows.Forms.Message m, System.Windows.Forms.MouseButtons button, int clicks)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.ButtonBase.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Button.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.OnMessage(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.Callback(System.IntPtr hWnd, int msg, System.IntPtr wparam, System.IntPtr lparam)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(System.IntPtr dwComponentID, int reason, int pvLoopData)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoop(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.RunDialog(System.Windows.Forms.Form form)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Form.ShowDialog(System.Windows.Forms.IWin32Window owner)	Unknown
 	GitUI.dll!GitUI.CommandsDialogs.BrowseDialog.FormOpenDirectory.OpenModule(System.Windows.Forms.IWin32Window owner, GitCommands.GitModule currentModule) Line 90	C#
 	GitUI.dll!GitUI.CommandsDialogs.BrowseDialog.DashboardControl.Dashboard.openItem_Click(object sender, System.EventArgs e) Line 231	C#
 	System.Windows.Forms.dll!System.Windows.Forms.Control.OnClick(System.EventArgs e)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.WmMouseUp(ref System.Windows.Forms.Message m, System.Windows.Forms.MouseButtons button, int clicks)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Label.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.LinkLabel.WndProc(ref System.Windows.Forms.Message msg)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.OnMessage(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.ControlNativeWindow.WndProc(ref System.Windows.Forms.Message m)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.Callback(System.IntPtr hWnd, int msg, System.IntPtr wparam, System.IntPtr lparam)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(System.IntPtr dwComponentID, int reason, int pvLoopData)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoop(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.Run(System.Windows.Forms.Form mainForm)	Unknown
 	GitExtensions.exe!GitExtensions.Program.RunApplication() Line 158	C#
 	GitExtensions.exe!GitExtensions.Program.Main() Line 73	C#

```
</details>

**Comment by gerhardol** [comment]
@RussKie 
Stack trace seem to be missing for native transitions. Can that be included?

Not so much can be said from the trace though, the exception is in WinForms code.
With .NET 5 the source is available at least, not sure if this can be debugged still.

**Comment by NikolayXHD** [comment]
My work PC which has OS updated years ago does reproduce the issue.

The trouble is, when I intercept the exception in the debugger the whole RDP session to work PC hangs to death until I log-out/log-in.

I guess I'll have to dance around and use my imagination a lot to figure what's going on.

**Comment by RussKie** [comment]
> Stack trace seem to be missing for native transitions. Can that be included?

tl;dr: no

`[Native to Managed Transition]` is a frame that indicates transition from the managed code to unmanaged and vice versa. You wouldn't get unmanaged traces unless you debug with native support turned on. And then you need to have debug symbols to have a meaningful stack trace.
